src/Features/Core/Diagnostics/EngineV1/DiagnosticAnalyzerDriver.cs

```diff
@@ -309,7 +309,7 @@ private IEnumerable<Diagnostic> GetFilteredDocumentDiagnosticsCore(IEnumerable<D
 
             return compilation == null
                 ? diagnostics
-                : CompilationWithAnalyzers.GetEffectiveDiagnostics(diagsFilteredByLocation, compilation);
+                : CompilationWithAnalyzers.GetEffectiveDiagnostics(diagsFilteredByLocation.ToImmutableArray(), compilation);
```

Comment:
Here we know that the value is non-null. I used to ImmutableArrayOrEmpty within the test extension method as I don't know if some caller passes in null.